pkg/routes/route_sync.go

```diff
@@ -1,82 +0,0 @@
-package routes
```

Comment:
@twz123 I ended up bringing this back into the `routing` package because I realized that it was behaving a bit more like an extension of the NetworkRoutesController, in that, in order to have it process route reconciliation I needed it to work with the routes source of truth, which is BGP. At that point it is no longer strictly related to host routes (although now that I'm thinking about it, it means that maybe `host_route_sync.go` isn't the right name for the file any more either.

Anyway, curious about your thoughts here as well.